src/jsonrpc.js

```diff
@@ -112,8 +160,20 @@ export default class JsonRpc {
         } else {
           const block = await bluebird.promisify(this.web3Instance.getBlock)(
             this.currentBlock, true);
-          await this._scanBlockCallback(block);
 
+          if (isFastMode) {
+            const logs = await this.getLogsFromOneBlock();
+            const blockTransactionsWithLogsList =
+            this.getBlockAndTransactionLogsFormat(block, logs);
+
+            if (this.callback) {
+              blockTransactionsWithLogsList.forEach((transaction) => {
+                this.callback(transaction);
+              });
+            }
+          } else {
+            await this._scanBlockCallback(block);
```

Comment:
Can we put whats inside this condition in a method just to make the code smaller here, with that there will be no need for the comments `end if` `end catch` `end loop`